msprime/intervals.py

```diff
@@ -449,50 +449,151 @@ def asdict(self):
+    fileobj,
+    sequence_length=None,
+    *,
+    has_header=True,
+    position_col=None,
+    rate_col=None,
+    map_col=None,
+):
     # Black barfs with an INTERNAL_ERROR trying to reformat this docstring,
     # so we explicitly disable reformatting here.
     # fmt: off
     """
     Parses the specified file in HapMap format. These files must be
-    white-space-delimited, and contain a single header line (which is
-    ignored), and then each subsequent line contains the starting position
-    and recombination rate for the segment from that position (inclusive)
-    to the starting position on the next line (exclusive). Starting
-    positions of each segment are given in units of bases, and
-    recombination rates in centimorgans/Megabase. The first column in this
-    file is ignored, as are additional columns after the third (Position is
-    assumed to be the second column, and Rate is assumed to be the third).
-    If the first starting position is not equal to zero, then a
-    zero-recombination region is inserted at the start of the chromosome.
-
-    A sample of this format is as follows::
-
-        Chromosome	Position(bp)	Rate(cM/Mb)	Map(cM)
-        chr1	55550	        2.981822	0.000000
-        chr1	82571	        2.082414	0.080572
-        chr1	88169	        2.081358	0.092229
-        chr1	254996	        3.354927	0.439456
-        chr1	564598	        2.887498	1.478148
+    white-space-delimited, and by default are assumed to contain a single header line
+    (which is ignored). Each subsequent line then contains a physical position (in base
+    pairs) and either a genetic map position (in centiMorgans) or a recombination rate
+    (in centiMorgans per megabase); the rate between the current physical position
+    (inclusive) and the physical position on the next line (exclusive) is taken as
+    constant. By default, the second column of the file is taken as the physical
+    position and the fourth column is taken as the genetic position, as seen in the
+    following sample of the format::
+
+        Chromosome	Position(bp)  Rate(cM/Mb)  Map(cM)
+        chr10       48232         0.1614       0.002664
+        chr10       48486         0.1589       0.002705
+        chr10       50009         0.159        0.002947
+        chr10       52147         0.1574       0.003287
         ...
-        chr1	182973428	2.512769	122.832331
-        chr1	183630013	0.000000	124.482178
-
-    :param str filename: The name of the file to be parsed. This may be
-        in plain text or gzipped plain text.
+        chr10	    133762002     3.358        181.129345
+        chr10	    133766368     0.000        181.144008
+
+    :: note..
+        If neither ``position_col``, ``rate_col``, or ``map_col`` are 0, then the
+        initial column is assumed to be the chromosome name, and a warning given if
+        rows do not contain identical names.
+
+    :param str fileobj: File, filename, or generator to read. If the filename
```

Comment:
I just copied the text basically verbatim from the `loadtxt` docs (as this is passed straight on to `numpy.loadtxt()`), rather than make the decision myself. But happy to remove this bit: as you say, it complicates the matter.